Output retry count for debugging purposes

louis-z opened the pull request:
Output retry count for debugging purposes
- GitLabTestContext: output retry count + remove unused code
- Add Assert in CreateMergeRequest to validate the default branch exists before carrying on

meziantou commented on NGitLab.Tests/Docker/GitLabTestContext.cs:
```diff
@@ -18,7 +19,13 @@ namespace NGitLab.Tests.Docker
+        private static readonly Policy s_gitlabRetryPolicy = Policy.Handle<GitLabException>()
+            .WaitAndRetry(_maxRetryCount, _ => TimeSpan.FromSeconds(1), (exception, timespan, retryCount, context) =>
+            {
+                TestContext.Progress.WriteLine($"{exception.Message} -> Polly Retry {retryCount} of {_maxRetryCount}...");
```
You should use TestContext.Out / TestContext.WriteLine to associate the log to the test as we don't need to see the message immediatly in the console.

louis-z replied:
Made use of `TestContext.WriteLine`. I also output the current test's full name.

louis-z committed:
Output retry count for debugging purposes
- GitLabTestContext: output retry count + remove unused code
- Add Assert in CreateMergeRequest to validate the default branch exists before carrying on

louis-z committed:
Use TestContext.Out instead of TestContext.Progress

louis-z committed:
Output test name too

louis-z committed:
Add more asserts in CreateMergeRequest
